### ISSUE TITLE
Filenames converted to lowercase

### ISSUE THREAD

**Issue by jhuebel** [notepad-plus-plus/notepad-plus-plus#710]
Sometimes, when a file is loaded from a remote server via SFTP, the filename of the cached copy of the file is converted to lowercase. So after the file is modified and saved back to the server, instead of overwriting the original file with a mixed-case filename, a new file is create with a filename in all lower case.

Steps to reproduce:

- Create a file on a Linux server called "Thisfile.php".
- On a Windows workstation, connect to the Linux server using NppFTP with the SFTP protocol.
- Open "Thisfile.php".
- Note that the cached copy of the file will be called "thisfile.php", which is not correct.
- Modify the file in some way.
- Save the file, which writes the changes back to the server.
- Instead of "Thisfile.php" being updated with the changes you just made, a new file will appear called "thisfile.php" in the same directory.

This only happens when the filename is a single word followed by a file extension. If the file is named "This_file.php" then case is preserved like it should be.

Severity:

This is a significant problem for developers who use CodeIgniter 3.0, since the Controllers are all named "Controller.php", where "Controller" is typically a single-word name for your controller in proper-case.

**Comment by jhuebel** [comment]
Figured it out. This may still be considered a bug, but it turns out that it occurs only when there are old cache files left over in NppFTP's cache directory.

I recently moved from CodeIgniter 2.x to 3.x. That included renaming files from "thisfile.php" to "Thisfile.php". Since there was an old cached copy of "thisfile.php" still on my computer (perhaps from a crashed session or something, when NppFTP downloaded a new copy of the renamed file ("Thisfile.php"), Windows treated it as updating the cached copy ("thisfile.php").

Basically, this boils down to Windows being stupid about case sensitive file naming. Windows really treats "Thisfile.php" and "thisfile.php" has the same filename, even though other (much more intelligent) operating systems honor the case sensitive naming and treat them as two different filenames.

**Comment by milipili** [comment]
Is this issue only a problem on NppFTP ?

**Comment by jhuebel** [comment]
It is limited to NppFTP. It's further limited to opening a remote file from a filesystem that properly handles case-sensitive file naming (Unix variants). So, I doubt this affects opening a file from a Windows machine (local or remote). Windows only superficially supports case sensitive filenames.

**Comment by dail8859** [comment]
Closing since this is 3rd party plugin related.
